### DISCUSSION TITLE
[@types/verror] error TS2416: Property 'cause' in type 'VError' is not assignable to the same property in base type 'Error'.

### DISCUSSION THREAD

**Discussion by marcushultman** [DefinitelyTyped/DefinitelyTyped#59121] [Issues with a @types package]
When targeting `es2022`/`esnext` the `Error` class defines the `cause` field to be of type `Error`. The `VError` type inherits from `Error` and thereby changes type of field `cause`, causing builds targeting the aforementioned versions to fail with the following build error:

```
node_modules/@types/verror/index.d.ts:31:5 - error TS2416: Property 'cause' in type 'VError' is not assignable to the same property in base type 'Error'.
  Property 'message' is missing in type '() => Error | undefined' but required in type 'Error'.

31     cause(): Error | undefined;
       ~~~~~

  node_modules/typescript/lib/lib.es5.d.ts:1023:5
    1023     message: string;
             ~~~~~~~
    'message' is declared here.

node_modules/@types/verror/index.d.ts:31:5 - error TS2425: Class 'Error' defines instance member property 'cause', but extended class 'VError' defines it as instance member function.

31     cause(): Error | undefined;
       ~~~~~
```

Versions:
`typescript`: 4.6.2
`@types/verror`: 1.10.5

[a comment by an automated account is not shown]

**Comment by schmod** [comment]
Duplicate of https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/59093

**Comment by schmod** [comment]
This needs to be fixed, but a workaround is to ensure that `esnext` or `es2022` are not present in your `tsconfig.json`'s `lib` array.

**Comment by voxpelli** [comment]
The first part of this will be fixed in TS 4.8 (https://github.com/microsoft/TypeScript/pull/49639) and doesn't only affect `@types/verror` but also polyfills for the new Error Cause standard (which supersedes `VError`) such as mine: https://github.com/voxpelli/pony-cause/issues/35

The second part still needs a fix in `@types/verror` though, I opened this one for that: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61199
